jina/serve/executors/decorators.py

```diff
@@ -104,15 +115,22 @@ def __init__(self, fn):
             if iscoroutinefunction(fn):
 
                 @functools.wraps(fn)
-                async def arg_wrapper(*args, **kwargs):
-                    return await fn(*args, **kwargs)
+                async def arg_wrapper(
+                    self2, *args, **kwargs
+                ):  # we need to get the summary from the executor, so we need to access the self
+                    with self._get_summary(self2._summary_method, fn.__name__):
+                        return await fn(self2, *args, **kwargs)
 
                 self.fn = arg_wrapper
             else:
 
                 @functools.wraps(fn)
-                def arg_wrapper(*args, **kwargs):
-                    return fn(*args, **kwargs)
+                def arg_wrapper(
+                    self2, *args, **kwargs
```

Comment:
maybe lets rename self2 to sth else? Is this conceptually a `self`? Its just `executor_instance` or sth no??